include/nuttx/ioexpander/gpio.h

```diff
@@ -293,7 +295,7 @@ int gpio_pin_unregister_byname(FAR struct gpio_dev_s *dev,
 #ifdef CONFIG_GPIO_LOWER_HALF
 struct ioexpander_dev_s;
 int gpio_lower_half(FAR struct ioexpander_dev_s *ioe, unsigned int pin,
-                    enum gpio_pintype_e pintype, int minor);
+                    enum gpio_pintype_e pintype, int minor, char *name);
```

Comment:
should we add a new api:
```
int gpio_lower_halfby_name(FAR struct ioexpander_dev_s *ioe, unsigned int pin,
                                             enum gpio_pintype_e pintype, int minor);
                                             enum gpio_pintype_e pintype, int minor, char *name);
```
like gpio_pin_register_byname to keep the compatibility.

Reply:
@xiaoxiang781216 I think using a single function in this case is fine, it avoids adding a new function to do same thing.

Reply:
consistency > simple

Reply:
@xiaoxiang781216 I think I agree with the consintency and that it should be under `gpio_lower_half_byname()`. But how about also introducing something like `static gpio_lower_half_internal()` that would basically wrap lines +- 392 to 431 in an internal function. This way we would avoid repeating the same code. Then we would have

```C
int gpio_lower_half(FAR struct ioexpander_dev_s *ioe, unsigned int pin,
                    enum gpio_pintype_e pintype, int minor)
{
  some variables init

  gpio_lower_half_internal(some args);

  ret = gpio_pin_register(gpio, minor);
}
```

and

```C
int gpio_lower_half(FAR struct ioexpander_dev_s *ioe, unsigned int pin,
                    enum gpio_pintype_e pintype, char *name)
{
  some variables init

  gpio_lower_half_internal(some args);

  ret = gpio_pin_register_byname(gpio, name);
}
```

Reply:
Yes, it looks good.

Reply:
Updated.